include/bout/invert_laplace.hxx

```diff
@@ -308,9 +312,23 @@ protected:
   int extra_yguards_lower; ///< exclude some number of points at the lower boundary, useful for staggered grids or when boundary conditions make inversion redundant
   int extra_yguards_upper; ///< exclude some number of points at the upper boundary, useful for staggered grids or when boundary conditions make inversion redundant
 
-  int global_flags;         ///< Default flags
-  int inner_boundary_flags; ///< Flags to set inner boundary condition
-  int outer_boundary_flags; ///< Flags to set outer boundary condition
+  /// Return true if global/default \p flag is set
+  bool isGlobalFlagSet(int flag) const { return (global_flags & flag) != 0; }
+  /// Return true if \p flag is set for the inner boundary condition
+  bool isInnerBoundaryFlagSet(int flag) const {
+    return (inner_boundary_flags & flag) != 0;
+  }
+  /// Return true if \p flag is set for the outer boundary condition
+  bool isOuterBoundaryFlagSet(int flag) const {
+    return (outer_boundary_flags & flag) != 0;
+  }
+
+  /// Return true if \p flag is set for the inner boundary condition
+  /// on the inner processor(s)
+  bool isInnerBoundaryFlagSetOnFirstX(int flag) const;
+  /// Return true if \p flag is set for the outer boundary condition
+  /// on the outer processor(s)
+  bool isOuterBoundaryFlagSetOnLastX(int flag) const;
```

Comment:
It sounds like this gets the flag from the first X proc, and uses it here.
```suggestion
  /// Return true if \p flag is set for the inner boundary condition
  /// and this is the first proc in X direction
  bool isInnerBoundaryFlagSetOnFirstX(int flag) const;
  /// Return true if \p flag is set for the outer boundary condition
  /// and this the last proc in X direction
  bool isOuterBoundaryFlagSetOnLastX(int flag) const;
```

Reply:
Thanks!